test/krew_test.go

```diff
@@ -36,13 +60,142 @@ func TestKrewInstall(t *testing.T) {
 	test.Call(validPlugin, "--help").RunOrFail()
 }
 
-func TestKrewHelp(t *testing.T) {
+func TestKrewUninstall(t *testing.T) {
 	skipShort(t)
 
 	test, cleanup := krew.NewTest(t)
 	defer cleanup()
 
-	test.Krew("help").RunOrFail()
+	test.WithIndex().Krew("install", validPlugin).RunOrFailOutput()
+	test.Krew("uninstall", validPlugin).RunOrFailOutput()
+	if err := test.Call(validPlugin, "--help").Run(); err == nil {
+		t.Errorf("Expected the plugin to be uninstalled")
+	}
+}
+
+func TestKrewSearchAll(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	output := test.WithIndex().Krew("search").RunOrFailOutput()
+	if plugins := lines(output); len(plugins) < 10 {
+		// the first line is the header
+		t.Errorf("Expected at least %d plugins", len(plugins)-1)
+	}
+}
+
+func TestKrewSearchOne(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := krew.NewTest(t)
+	defer cleanup()
+
+	plugins := lines(test.WithIndex().Krew("search", "krew").RunOrFailOutput())
+	if len(plugins) < 2 {
+		t.Errorf("Expected krew to be a valid plugin")
+	}
+	if !strings.HasPrefix(plugins[1], "krew ") {
+		t.Errorf("The first match should be krew")
+	}
+}
+
+func TestKrewInfo(t *testing.T) {
+	skipShort(t)
+
+	tests := []struct {
+		name      string
+		plugin    string
+		shouldErr bool
+	}{
+		{
```

Comment:
please don't put negative cases to the same test case.
they are different test cases. and these aren't unit tests. tabular tests like []struct{}{} are good for tabular unit tests. but ideally each integration test case should be a separate method testing a particular scenario workflow.

e.g. try

TestKrewInfo
TestKrewInfoInvalidPlugin

Reply:
Makes sense. The test cases are not split into separate test functions.